statsmodels/base/distributed_estimation.py

```diff
@@ -0,0 +1,545 @@
+        A generator that produces a sequence of tuples where the first
+        element in the tuple corresponds to an endog array and the
+        element corresponds to an exog array.
+    partitions : scalar
+        The number of partitions that the data will be split into.
+    model_class : statsmodels model class
+        The model class which will be used for estimation. If None
+        this defaults to OLS.
+    init_kwds : dict-like or None
+        Keywords needed for initializing the model, in addition to
+        endog and exog.
+    fit_kwds : dict-like or None
+        Keywords needed for the model fitting.
+    estimation_method : function or None
+        The method that performs the estimation for each partition.
+        If None this defaults to _est_regularized_debiased.
+    estimation_kwds : dict-like or None
+        Keywords to be passed to estimation_method.
+    join_method : function or None
+        The method used to recombine the results from each partition.
+        If None this defaults to _join_debiased.
+
+    Attributes
+    ----------
+    data_generator : generator
+        See Parameters.
+    partitions : scalar
+        See Parameters.
+    model_class : statsmodels model class
+        See Parameters.
+    init_kwds : dict-like or None
+        See Parameters.
+    fit_kwds : dict-like or None
+        See Parameters.
+    estimation_method : function or None
+        See Parameters.
+    estimation_kwds : dict-like or None
+        See Parameters.
+    join_method : function or None
+        See Parameters.
+
+    Examples
+    --------
+
+    Notes
+    -----
+    """
+
+    def __init__(self, data_generator, partitions, model_class=None,
+                 init_kwds=None, fit_kwds=None, estimation_method=None,
+                 estimation_kwds=None, join_method=None,
+                 join_kwds=None):
+
+        self.data_generator = data_generator
+        self.partitions = partitions
+
+        if model_class is None:
+            self.model_class = OLS
+        else:
+            self.model_class = model_class
+
+        if init_kwds is None:
+            self.init_kwds = {}
+        else:
+            self.init_kwds = init_kwds
+
+        if fit_kwds is None:
+            self.fit_kwds = {}
+        else:
+            self.fit_kwds = fit_kwds
+
+        if estimation_method is None:
+            self.estimation_method = _est_regularized_debiased
+        else:
+            self.estimation_method = estimation_method
+
+        if estimation_kwds is None:
+            self.estimation_kwds = {}
+        else:
+            self.estimation_kwds = esitmation_kwds
+
+        if join_method is None:
+            self.join_method = _join_debiased
+        else:
+            self.join_method = join_method
+
+        if join_kwds is None:
+            self.join_kwds = {}
+        else:
+            self.join_kwds = join_kwds
+
+
+    def fit_distributed(self, parallel_method="sequential",
```

Comment:
Since the class name already has "Distributed" this can just be "fit" for brevity.